rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -244,3 +244,110 @@ def draw_patches(img, idx, bboxes):
+                 change during execution. If True, the pytorch tensor will be resized accordingly
+                 if the shape of RALI returned tensors changes during execution.
+                 If False, the iterator will fail in case of change.
+    last_batch_padded : bool, optional, default = False
+                 Whether the last batch provided by RALI is padded with the last sample
+                 or it just wraps up. In the conjunction with `fill_last_batch` it tells
+                 if the iterator returning last batch with data only partially filled with
+                 data from the current epoch is dropping padding samples or samples from
+                 the next epoch. If set to False next epoch will end sooner as data from
+                 it was consumed but dropped. If set to True next epoch would be the
+                 same length as the first one.
+
+    Example
+    -------
+    With the data set [1,2,3,4,5,6,7] and the batch size 2:
+    fill_last_batch = False, last_batch_padded = True  -> last batch = [7], next iteration will return [1, 2]
+    fill_last_batch = False, last_batch_padded = False -> last batch = [7], next iteration will return [2, 3]
+    fill_last_batch = True, last_batch_padded = True   -> last batch = [7, 7], next iteration will return [1, 2]
+    fill_last_batch = True, last_batch_padded = False  -> last batch = [7, 1], next iteration will return [2, 3]
+    """
+    def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = False, multiplier = [1.0, 1.0, 1.0], offset = [0.0, 0.0, 0.0], tensor_dtype = types.FLOAT, size = -1, auto_reset = False, device = "cpu", device_id = 0):
+        self.loader = pipeline
+        self.tensor_format = tensor_layout
+        self.multiplier = multiplier
+        self.offset = offset
+        self.reverse_channels = reverse_channels
+        self.tensor_dtype = tensor_dtype
+        self.device = device
+        self.device_id = device_id
+        self.output = None
+        self.len = b.getRemainingImages(self.loader._handle)
+        # self.last_batch_policy = self.loader._last_batch_policy #commented for now
+        self.shard_size = size
+        self.auto_reset = auto_reset
+        self.batch_count = 0
+        self.audio_length = None
+        self.samples = None
+        self.channels = None
+        self.max_shape = None
+        self.batch_size = self.loader._batch_size
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.loader.rocalRun() != 0:
+            raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.getOutputTensors()
+
+        self.batch_count = self.batch_count + self.batch_size
+        self.num_of_dims = self.output_tensor_list[0].num_of_dims()
+        if self.num_of_dims == 3:
+            self.batch_size = self.output_tensor_list[0].batch_size() if self.batch_size is None else self.batch_size
```

Comment:
The self.batch_size = self.loader._batch_size is already set in init. This can be removed